scripts/pre-commit.sh

```
@@ -0,0 +1,343 @@
+  fi
+  
+  echo -e "\n${BOLD}${BLUE}Recent error logs:${NC}"
+  
+  # List 10 most recent error logs
+  local i=1
+  local log_files=()
+  
+  while IFS= read -r log_file; do
+    log_files+=("$log_file")
+    # Extract step name and timestamp from filename
+    local filename=$(basename "$log_file")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)